### PR TITLE
For #18837: Settings menu item should be tinted if there is an account issue

### PR DESCRIPTION
**Pull request by eliserichards**
For #18837

> To reproduce this case for testing, you can pair with your device, then go to https://accounts.firefox.com/settings to manage your devices and remove the device from the account console. Re-open Fenix and you should be in the re-auth state.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -490,7 +490,9 @@ open class DefaultToolbarMenu(
     val settingsItem = BrowserMenuHighlightableItem(
         label = context.getString(R.string.browser_menu_settings),
         startImageResource = R.drawable.ic_settings,
-        iconTintColorResource = primaryTextColor(),
+        iconTintColorResource = if (hasAccountProblem)
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:493]: See line 203 (the old settings item) has this same tinting. We missed that in the new item implementation

**Comment by jonalmeida** [comment]
Curious what UX thinks. I mentioned we would move the re-auth icon on the left as well.

**Review by gabrielluong — APPROVED** [review]
Maybe we should go with the approach you took in https://github.com/mozilla-mobile/fenix/pull/18850 where we can move the `settingsItem` variable out of both of the old/new menu so that they can be reused in both since they are identical.

**Review by kglazko — APPROVED** [review]
Looks good to me after testing. Let's see what UX says and then merge it if all good!

**Comment by eliserichards** [comment]
This issue was to add the pre-existing auth error icons, so I think we can file a new issue for any future UX changes to this design

**Comment by jonalmeida** [comment]
@eliserichards I think there was a misunderstanding - this issue was to move the highlighting to the the 'Signed in as [email]' highlight.

Seems a bit awkward to highlight the 'Settings' that Sync has an issue, right?

**Comment by eliserichards** [comment]
I discussed the change with Victoria and she wants the icon to stay in settings for now (which is the pre-existing functionality). She doesn't have the bandwidth to redesign the sign in button now. That will be post-MR1 work.

**Pull request merged**